packages/utils/odsp-doclib-utils/src/odspErrorUtils.ts

```diff
@@ -124,15 +126,36 @@ export function createOdspNetworkError(
     props: ITelemetryProperties = {},
 ): LoggingError & OdspError & IFacetCodes {
     let error: LoggingError & OdspError & IFacetCodes;
+    const parseResult = tryParseErrorResponse(responseText);
+    let facetCodes: string[] | undefined;
+    let innerMostErrorCode: string | undefined;
+    if (parseResult.success) {
+        // Log the whole response if it looks like the error format we expect
+        props.response = responseText;
+        const errorResponse = parseResult.errorResponse;
+        facetCodes = parseFacetCodes(errorResponse);
+        if (facetCodes !== undefined) {
+            innerMostErrorCode = facetCodes[0];
+            props.innerMostErrorCode = innerMostErrorCode;
+        }
+    }
     switch (statusCode) {
         case 400:
             error = new GenericNetworkError(errorMessage, false, { statusCode });
             break;
         case 401:
         case 403:
-            const claims = response?.headers ? parseAuthErrorClaims(response.headers) : undefined;
-            const tenantId = response?.headers ? parseAuthErrorTenant(response.headers) : undefined;
-            error = new AuthorizationError(errorMessage, claims, tenantId, { statusCode });
+            if (innerMostErrorCode === OdspErrorType.serviceReadOnly) {
```

Comment:
I find it odd to use same type / name for ODSP internal errors and for our public API surface. I'd prefer these to be two different things. I.e. 

const OdspServiceReadOnlyErrorCode = "...";

enum OdspErrorType {
... 
serviceReadOnly = "";
}

Their values could be the same, but I'd rather not collide them to remove temptation to leverage one where the other one needs to be used.

That said, I thought initial messaging was to use headers to detect it, no? Maybe guidance changed...

Reply:
Yes, the guidance changed on how to detect the error. No headers now.

Reply:
That's fine, but we should not compare innerMostErrorCode  to OdspErrorType.serviceReadOnly. Developer should be free to change OdspErrorType enums and the code should continue to work as expected (minus host changes). There should be clearly defined constant for this value with a comment that the value is defined by ODSP and can't be changed.

Reply:
I added that constant in this change. Will add this comment too.